dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AnnotationLiteralsTest.war/src/io/openliberty/org/apache/faces40/fat/literals/EventBean.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.org.apache.faces40.fat.literals;
+
+import java.io.Serializable;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.enterprise.util.TypeLiteral;
+import jakarta.faces.application.FacesMessage;
+import jakarta.faces.context.FacesContext;
+import jakarta.faces.event.WebsocketEvent;
+import jakarta.faces.push.Push;
+import jakarta.faces.push.PushContext;
+import jakarta.inject.Named;
+
+@Named(value = "eventBean")
+@ApplicationScoped
+public class EventBean implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    public static final String channelName = "testChannel";
+
+    public void logOpenWebsocket() {
+        WebsocketEvent opened = CDI.current().select(new TypeLiteral<WebsocketEvent>() {}, WebsocketEvent.Opened.Literal.INSTANCE).get();
+        String message = "WebsocketEvent.Opened=" + Boolean.valueOf(opened != null && opened.getChannel().equals(channelName));
+        FacesContext.getCurrentInstance().addMessage("open:messages", new FacesMessage(message));
```

Comment:
Messages don't display for me, and I think it's because of the ids. 

https://jakarta.ee/specifications/faces/4.0/apidocs/jakarta/faces/context/facescontext#addMessage(java.lang.String,jakarta.faces.application.FacesMessage)

open:messages is the clientId, but you just list <h:messages id="messages" />

Let me double check

Reply:
I think there's a bigger problem here.  The eventBean bean isn't invoked with 
```
<h:commandButton id="submit" value="Log Event" action="#{eventBean.logOpenWebsocket()}" >
  <f:ajax/>
</h:commandButton>
```

Reply:
I don't think this is valid scenario, but I spec issue should be opened 

Calling `bsocketEvent opened = CDI.current().select(new TypeLiteral<WebsocketEvent>() {}, WebsocketEvent.Opened.Literal.INSTANCE).get();`

Gives me the following error on MyFaces and Mojarra
```
Caused by: org.jboss.weld.exceptions.UnsatisfiedResolutionException: WELD-001334: Unsatisfied dependencies for type WebsocketEvent with qualifiers @Opened 
```

Reply:
WebsocketEvent is just a class -- It doesn't look to interact much with CDI?